### ISSUE TITLE
Order not retained for api_stages in aws_api_gateway_usage_plan

### ISSUE THREAD

**Issue by archil620** [hashicorp/terraform#28323]
### Terraform Version
v12.25

### Situation
I'm using an aws_api_gateway_usage_plan which applies to multiple stages, defined with the api_stages config blocks.

### Terraform Configuration Files 
**Module main.tf file**
```
resource "aws_api_gateway_usage_plan" "usage_plan" {
  name         = var.usage_plan_name
  dynamic "api_stages" {
      for_each = var.api_stages
      content {
        api_id     = api_stages.value.api_id
        stage = api_stages.value.stage      
      }
    }
  throttle_settings {
    burst_limit = var.burst_limit
    rate_limit  = var.rate_limit
  }
  description = var.description
  tags = var.app_role_tags
}
```
**Module variable.tf file** Only providing variable relevant to this issue
```
variable "api_stages" {
  type = set(object({
    api_id    = string
    stage = string
  }))
}
```
Now the resource where this module is used

**Resource main.tf file**

```
module "SCCG_Usage_Plan" {
  source           = "/modulepath"
  usage_plan_name        = "usageplan name"
  burst_limit = 5
  rate_limit = 10
  api_stages = var.api_stages
  app_role_tags = var.app_role_tags
  description = "Desctiption"
}
```

Variable.tf file same as module one

I am passing api_id and stages, through **terraform.tfvars** file

```
api_stages={
  {
      api_id= "9afgfdg"
      stage="sbx"
  },
  {
      api_id= "ukxydgdgd"
      stage="sbx"
  },
  {
      api_id= "8pa5gdgdg"
      stage="sbx"
  },
  {
      api_id= "zo0tkadad"
      stage="sbx"
  }  
}
```

Now whenever I have new API, i'll add api_id and stages here, and this will add those info to usage plan, works fine.

Issue occurs when I try to add any api_id or stages that doesnt exist. at that time,  i'll give error like

```
Error: error updating API Gateway Usage Plan: NotFoundException: API Stage not found: zo0tk1nhug:sandbox

  on .terraform/modules/main.tf line 1, in resource "aws_api_gateway_usage_plan" "usage_plan":
   1: resource "aws_api_gateway_usage_plan" "usage_plan" {

```

And that's fine, but after removing the faulty api_id/stages, when I re-run the plan, i see other api_ids those were already added and running fine deleted from the usage plan, and terraform is trying to add those again, 

### Actual Behavior
```
 # module.Usage_Plan.aws_api_gateway_usage_plan.usage_plan will be updated in-place
  ~ resource "aws_api_gateway_usage_plan" "usage_plan" {
        arn         = "arn:aws:apigateway:us-east-1::/usageplans/xxxxx"
        description = "Desctiption
        id          = "xxxxx"
        name        = "usageplan"
        
      + api_stages {
          + api_id = "8pa5gdgdg"
          + stage  = "sandbox"
        }
        api_stages {
            api_id = "9afgfdg"
            stage  = "sandbox"
        }
      + api_stages {
          + api_id = "ukxydgdgd"
          + stage  = "sandbox"
        }
      
        throttle_settings {
            burst_limit = 5
            rate_limit  = 10
        }
    }

As you can see it's randomly deleting api_id and stages if error occurs and trying to add those back in next apply
```

### Expected Behavior
```
 # module.Usage_Plan.aws_api_gateway_usage_plan.usage_plan will be updated in-place
  ~ resource "aws_api_gateway_usage_plan" "usage_plan" {
        arn         = "arn:aws:apigateway:us-east-1::/usageplans/xxxxx"
        description = "Desctiption
        id          = "xxxxx"
        name        = "usageplan"
        
       api_stages {
           api_id = "8pa5gdgdg"
           stage  = "sandbox"
        }
        api_stages {
            api_id = "9afgfdg"
            stage  = "sandbox"
        }
       api_stages {
           api_id = "ukxydgdgd"
           stage  = "sandbox"
        }
            throttle_settings {
            burst_limit = 5
            rate_limit  = 10
        }
    }

Basically no changes as these  api_id and stages are already there
```
### References
https://github.com/hashicorp/terraform-provider-aws/issues/1675

**Comment by mildwonkey** [comment]
Hi @archil620 , sorry you are experiencing this unwanted behavior!

The linked issue indicates this was an AWS issue which was addressed in the 3.13.0 release of the AWS provider. What version of the AWS provider are you using currently? If it is 3.13.0 or later, you may want to open a new issue in the AWS repository. Otherwise, you will need to upgrade your AWS provider version in order to get past this problem. 

There doesn't appear to be a core issue here, so I am going to close this issue. Thanks!

**Comment by archil620** [comment]
I am using aws provider version 3.15
